testbed/tests/app/test_desktop.py

```diff
@@ -354,73 +361,74 @@ async def test_presentation_mode_exit_on_window_state_change(
         "App is not in presentation mode" f"\nTest Window 1 is in {new_window_state}",
-        minimize=True if new_window_state == WindowState.MINIMIZED else False,
-        full_screen=True if new_window_state == WindowState.FULLSCREEN else False,
+        state=new_window_state,
     )
 
     assert not app.in_presentation_mode
     assert window1_probe.instantaneous_state == new_window_state
 
     # Reset window states
     window1.state = WindowState.NORMAL
+    # Wait for window animation before assertion.
+    await window1_probe.wait_for_window(
+        "All test windows are in WindowState.NORMAL", state=WindowState.NORMAL
+    )
     window2.state = WindowState.NORMAL
-    # Add delay to ensure windows are visible after animation.
-    await main_window_probe.wait_for_window(
-        "All test windows are in WindowState.NORMAL",
-        minimize=True if new_window_state == WindowState.MINIMIZED else False,
-        full_screen=True if new_window_state == WindowState.FULLSCREEN else False,
+    # Wait for window animation before assertion.
+    await window2_probe.wait_for_window(
+        "All test windows are in WindowState.NORMAL", state=WindowState.NORMAL
     )
     assert window1_probe.instantaneous_state == WindowState.NORMAL
     assert window2_probe.instantaneous_state == WindowState.NORMAL
 
     # Enter presentation mode again
     app.enter_presentation_mode([window1])
-    # Add delay to ensure windows are visible after animation.
-    await main_window_probe.wait_for_window(
-        "App is in presentation mode",
-        minimize=True if new_window_state == WindowState.MINIMIZED else False,
-        full_screen=True if new_window_state == WindowState.FULLSCREEN else False,
+    # Wait for window animation before assertion.
+    await window1_probe.wait_for_window(
+        "App is in presentation mode", state=WindowState.PRESENTATION
     )
     assert app.in_presentation_mode
     assert window1_probe.instantaneous_state == WindowState.PRESENTATION
 
     # Changing window state of extra window should make the app exit presentation mode.
     window2.state = new_window_state
-    # Add delay to ensure windows are visible after animation.
-    await main_window_probe.wait_for_window(
+    # Wait for window animation before assertion.
+    await window2_probe.wait_for_window(
         "App is not in presentation mode" f"\nTest Window 2 is in {new_window_state}",
-        minimize=True if new_window_state == WindowState.MINIMIZED else False,
-        full_screen=True if new_window_state == WindowState.FULLSCREEN else False,
+        state=new_window_state,
     )
 
     assert not app.in_presentation_mode
     assert window2_probe.instantaneous_state == new_window_state
 
     # Reset window states
     window1.state = WindowState.NORMAL
+    # Wait for window animation before assertion.
+    await window1_probe.wait_for_window(
+        "All test windows are in WindowState.NORMAL", state=WindowState.NORMAL
+    )
```

Comment:
This assertion should be *after* the `window2.state` assignment, so that both windows can be restored in parallel, rather than running strictly in series.

Reply:
Thanks, I have changed it.